File: stacks/dart/web/test/greeter_test.dart
import 'package:test/test.dart';
import 'package:examples_dart_hello/greeter.dart';

void main() {
  test('default greeting', () {
    expect(const Greeter('World').greet(), 'Hello, World!');
  });

  test('custom prefix', () {
    expect(const Greeter('Ala').greet(prefix: 'Cześć'), 'Cześć, Ala!');
  });
}

File: stacks/backend/go/internal/greeter/greeter_test.go
package greeter

import "testing"

func TestGreet(t *testing.T) {
	if got := Greet("World"); got != "Hello, World!" {
		t.Fatalf("expected 'Hello, World!' got %q", got)
	}
	if got := Greet(""); got != "Hello, World!" {
		t.Fatalf("expected default to 'Hello, World!' got %q", got)
	}
}
